chrome/content/zotero/xpcom/style.js

```diff
@@ -757,19 +761,36 @@ Zotero.Style.prototype.getCiteProc = function(locale, automaticJournalAbbreviati
 	}
 	
 	try {
-		var citeproc = new Zotero.CiteProc.CSL.Engine(
-			new Zotero.Cite.System({
-				automaticJournalAbbreviations,
-				uppercaseSubtitles
-			}),
-			xml,
-			locale,
-			overrideLocale
-		);
-		
-		citeproc.opt.development_extensions.wrap_url_and_doi = true;
-		// Don't try to parse author names. We parse them in itemToCSLJSON
-		citeproc.opt.development_extensions.parse_names = false;
+		var citeproc;
+		if (Zotero.Prefs.get('cite.useCiteprocRs')) {
+			citeproc = new Zotero.CiteprocRs.Engine(
+				new Zotero.Cite.System({
+					automaticJournalAbbreviations,
+					uppercaseSubtitles: uppercaseSubtitles
+				}),
+				this,
+				xml,
+				locale,
+				format == 'text' ? 'plain' : format,
+				overrideLocale
+			);
+		}
+		else {
+			citeproc = new Zotero.CiteProc.CSL.Engine(
+				new Zotero.Cite.System({
+					automaticJournalAbbreviations,
+					uppercaseSubtitles
+				}),
+				xml,
+				locale,
+				overrideLocale
+			);
+			citeproc.setOutputFormat(format);
+			citeproc.free = () => 0;
+			citeproc.opt.development_extensions.wrap_url_and_doi = true;
+			// Don't try to parse author names. We parse them in itemToCSLJSON
+			citeproc.opt.development_extensions.parse_names = false;
```

Comment:
I actually have a name parser as well, do you need an option to disable it?

Reply:
Well, I guess it's about CSL-JSON output and uses beyond our built-in citation processing, right? Does the CSL spec call for specific name-parsing behavior, or was that just something that citeproc-js implemented? I assume the goal here was just generating CSL-JSON for consumers that might not have complex name parsers built-in.

Does disabling the name parser actually make a difference if the CSL-JSON is already pre-parsed?